includes/connect/class-wc-stripe-connect.php

```diff
@@ -255,6 +255,7 @@ private function save_stripe_keys( $result, $type = 'connect', $mode = 'live' )
 			$options['enabled']                         = 'yes';
 			$options['testmode']                        = $is_test ? 'yes' : 'no';
 			$options['upe_checkout_experience_enabled'] = $this->get_upe_checkout_experience_enabled();
+			$options['optimized_checkout_element']      = $this->get_optimized_checkout_element_default_value();
```

Comment:
I will keep thinking about this tomorrow, but it feels like it might be _really_ good to add some unit tests so we can correctly track what is going on here and what edge cases might exist.

I am concerned that we might have edge cases where merchants upgrade, and then reconnect, and we auto-enable Optimized Checkout just because we hadn't saved any data in that sub-option from before that reconnection attempt.

Might it be better to create some code in `WC_Stripe::install()` and/or hook into the `woocommerce_stripe_updated` action to set/pick the default value for optimized checkout based on whether there is data in the `wc_stripe_version` option from before? That way OCS would be default-on for fresh new installs only, whereas we'd leave it off for any other install.